tools/swig/CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+# CMake configuration for generating script language interfaces with SWIG
+
+# set minimum CMake version required and switch to new policies for SWIG
+cmake_minimum_required(VERSION 3.14)
+if(POLICY CMP0078)
+  cmake_policy(SET CMP0078 NEW)
+endif()
+if(POLICY CMP0086)
+  cmake_policy(SET CMP0086 NEW)
+endif()
+
+# some of the find_package() scripts trigger developer warnings
+# even though they are bundled with CMake; Suppress them
+set(CMAKE_SUPPRESS_DEVELOPER_WARNINGS TRUE)
+
+if(NOT BUILD_SHARED_LIBS)
+  error(FATAL_ERROR "Option BUILD_SHARED_LIBS must be enabled to use SWIG wrappers")
+endif()
+
+find_package(SWIG REQUIRED)
+include(${SWIG_USE_FILE})
+
+option(BUILD_SWIG_JAVA "Build Java JNI wrapper with SWIG" OFF)
+option(BUILD_SWIG_LUA "Build Lua wrapper with SWIG" OFF)
+option(BUILD_SWIG_PERL5 "Build Perl5 wrapper with SWIG" OFF)
+option(BUILD_SWIG_PYTHON "Build Python wrapper with SWIG" OFF)
+option(BUILD_SWIG_RUBY "Build Ruby wrapper with SWIG" OFF)
+option(BUILD_SWIG_TCL "Build Tcl wrapper with SWIG" OFF)
+
+add_custom_target(clean-all
+  COMMAND ${CMAKE_COMMAND} --build . --target clean
+  COMMAND ${CMAKE_COMMAND} -E remove example.* *.class *.java *.pm log.lammps
+  COMMAND ${CMAKE_COMMAND} -E remove_directory __pycache__)
+
+if(BUILD_SWIG_JAVA)
+  set_property(SOURCE lammps.i PROPERTY SWIG_MODULE_NAME javalammps)
+  swig_add_library(javalammps TYPE MODULE LANGUAGE java SOURCES lammps.i)
+  find_package(JNI REQUIRED)
+  target_include_directories(javalammps PRIVATE ${JNI_INCLUDE_DIRS})
+  swig_link_libraries(javalammps PRIVATE lammps ${JNI_LIBRARIES})
+  configure_file(run_java_example.sh run_java_example.sh COPYONLY)
```

Comment:
It would be nice to run this as a test as well.

Reply:
No way. This is the result of a weekend hack because of curiosity and thus is "take it or leave it" feature without the intention to maintain it. The docs clearly say it is incomplete and only partially tested. To make these proper interfaces, somebody has to step up, finish the implementation and agree to maintain it. It was surprising how much functionality can be made to work with rather little effort, but outside of a little polishing of the docs, and an occasional update of the `lammps.i` file after we make changes to `library.h` I am not going to invest much effort into this. There are other parts of LAMMPS that deserve more attention. To quote "Lawrence of Arabia": this is a sideshow of a sideshow!